source/beamline/CLS/CLSShutters.cpp

```diff
@@ -14,7 +19,88 @@ CLSShutters::CLSShutters(const QString &name, QObject *parent) :
 
 CLSShutters::~CLSShutters()
 {
+	shuttersBeamOnOrderMap_.clear();
+}
+
+AMListAction3* CLSShutters::createBeamOnActionList()
```

Comment:
Yes, I don't think it makes sense to move it out of CLSShutters. There is an existing implementation of `CLSShutters::createMoveToOpen/ClosedActions()` in this class that doesn't take the newly implemented order into account; I think @XuetaoLiu-CLS's code in `CLSShutters::createBeamOn/OffActionList()` could replace the older code.